vitessio/vitess
# Migrate the S3 SDK from v1 to v2

## Description

This PR updates the version of the AWS SDK from v1 to v2. The version 1 has been deprecated for some time now.

I have ran the following manual test:
1. Create a Vitess cluster (1 unsharded keyspace) from the `main` branch
1. Insert some data
1. Backup the cluster into AWS S3
1. Teardown the cluster
1. Create a Vitess cluster (same config) from the `migrate-to-s3-v2` branch (with `--restore_from_backup`)
1. Restore from the previous backup
1. Observe that the inserted data shows up after restoring
1. Take another backup on AWS S3
1. List all available backup on AWS S3
2. Restore the cluster from the latest backup (using `RestoreFromBackup`)
1. Remove the first backup we did (the one from `main`) from AWS S3
2. Teardown the cluster
3. Create another Vitess cluster (same config) from the `main` branch (with `--restore_from_backup`)
4. Observe that the inserted data shows up after restoring

This test ensures that we can upgrade: old backups will still work with the new code, and that we can downgrade: new backups will work with the old code. While making sure that no config/flag change is needed between the two code version. It also checks that the `GetBackups` and `RestoreFromBackup` vtctldclient commands work.

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required
